stdlib/utils/src/main/java/org/ballerinalang/utils/Next.java

```diff
@@ -86,6 +87,8 @@ public static Object next(Strand strand, Object iterator) {
     private static org.ballerinalang.jvm.types.BType getFiniteType(Object value) {
         Set<Object> valueSpace = new HashSet<>();
         valueSpace.add(value);
-        return new BFiniteType(value.toString(), valueSpace);
+        int masks = value == null ? TypeFlags.NILABLE : 0;
+        return new BFiniteType(value.toString(), valueSpace,
+                TypeFlags.addToMask(masks, TypeFlags.ANYDATA, TypeFlags.PURETYPE));
```

Comment:
Don't we have to get the type of the value and check if it is anydata/pure?

Reply:
We reach here only if the value is primitive. Hence it is safe to add these.